osi_route.proto

```diff
@@ -0,0 +1,106 @@
+// in the order they are listed.
+//
+// ## Example
+//
+// The example below shows the \link Route route\endlink of a vehicle.
+//
+// \image html OSI_Planned_Route.png "Route" width=850px
+//
+// The route is composed of three \link RoadSegment road segments\endlink RS1-3,
+// each indicated by a yellow outline. Two of the road segments
+// (RS2 and RS3) only contain a single \link LogicalLaneSegment logical lane segment\endlink
+// (highlighted in blue), while RS1 is composed of three
+// logical lane segments (green, blue and red).
+//
+message Route
+{
+
+    // Road segments that form the planned route of an agent.
+    //
+    // Consecutive segments should be connected without gaps, meaning that the
+    // two of them should form a continuous area.
+    //
+    repeated RoadSegment planned_route = 1;
+
+    //
+    // \brief A segment of a logical lane.
+    //
+    message LogicalLaneSegment
+    {
+        // The ID of the logical lane this segment belongs to.
+        //
+        // \rules
+        // refers_to: LogicalLane
+        // \endrules
+        //
+        optional Identifier logical_lane_id = 1;
+
+        // S position on the logical lane where the segment starts.
+        //
+        optional double start_s = 2;
+
+        // S position on the logical lane where the segment ends.
+        //
+        optional double end_s = 3;
+    }
+
+    //
+    // \brief A segment of a road.
+    //
+    // A road segment describes a segment of a road by listing the lanes that are
+    // in there. In general, only lanes should be listed that are parallel to each
+    // other, though some may have different lenghts (e.g. if a lane widening
+    // occurs, the newly appearing lane will have a shorter length).
+    //
+    // Typically a road segment will be either
+    // - a set of lanes between two junctions, or
+    // - parallel lanes on an intersection with the same driving direction
+    //
+    // ## Example
+    //
+    // Consider the \link RoadSegment road segment\endlink between two intersections,
+    // shown in the image below.
+    //
+    // \image html OSI_Road_Segment.png "RoadSegment" width=850px
+    //
+    // In the example, a single road segment RS with three
+    // \link LogicalLaneSegment logical lane segments\endlink LL1, LL2 and LL3 is
+    // shown. The segments are indicated by the green, blue and red highlighted areas,
+    // one for each underlying logical lane The starting
+    // s-position of each segment is indicated by the yellow dotted line and the s- prefix
+    // (note that the start of LL2 lies further to the left, outside of the image),
+    // while the ending s-position of all segments is shown by the yellow dotted line e-RS.
+    //
+    // As it can be seen in the example, all logical lane segments are parallel,
+    // but two of them are opening at a later position, so their starting
+    // s-positions will be different.
+    //
+    message RoadSegment
```

Comment:
Feedback from OSI Harmonization group: "Road" is an ambiguous term and should be avoided: "Road" can refer to an OpenDRIVE road (a technical term), or it can refer to what humans mean by road (which can span junctions, and for which there is probably no agreed-upon definition). Suggestion: use RouteSegment or similar name.

Reply:
I renamed the message and adjusted the description accordingly. I also slightly improved the documentation of the route segment as it wasn't written very well.